app/models/phylo_tree.rb

```diff
@@ -61,12 +64,29 @@ def job_command
     #     'NR': [first_byte, last_byte, source_s3_file, sample_id, align_viz_s3_file]
     #   },
     #   pipeline_run_id_2: ... }
-    taxon_byteranges = TaxonByterange.where(pipeline_run_id: pipeline_runs.pluck(:id)).where(taxid: taxid)
+    pipeline_run_ids = pipeline_runs.pluck(:id)
+    taxon_byteranges = TaxonByterange.where(pipeline_run_id: pipeline_run_ids).where(taxid: taxid)
     taxon_byteranges_hash = {}
     taxon_byteranges.each do |tbr|
       taxon_byteranges_hash[tbr.pipeline_run_id] ||= {}
       taxon_byteranges_hash[tbr.pipeline_run_id][tbr.hit_type] = [tbr.first_byte, tbr.last_byte]
     end
+    # If the tree is for genus level or higher, get the top species underneath (these species will have genbank records pulled in idseq-dag)
+    if tax_level > TaxonCount::TAX_LEVEL_SPECIES
+      level_str = (TaxonCount::LEVEL_2_NAME[tax_level]).to_s
+      taxid_column = "#{level_str}_taxid"
+      species_counts = TaxonCount.where(pipeline_run_id: pipeline_run_ids).where("#{taxid_column} = ?", taxid).where(tax_level: TaxonCount::TAX_LEVEL_SPECIES)
+      species_counts = species_counts.order('pipeline_run_id, count DESC')
+      top_taxid_by_run_id = {}
+      species_counts.each do |sc|
+        top_taxid_by_run_id[sc.pipeline_run_id] ||= sc.tax_id
+      end
+      reference_taxids = top_taxid_by_run_id.values
+    else
+      reference_taxids = [taxid]
+    end
+    # Retrieve superkigdom name for idseq-dag
+    superkingdom_name = TaxonLineage.where(taxid: 573).last.superkingdom_name
```

Comment:
What is 573 taxid? Can we add it to a constant for readability?

Reply:
oops that's a glitch, it should be `where(taxid: taxid)`, good catch